allennlp/commands/train.py

```diff
@@ -108,7 +109,10 @@ def train_model(param_dict: Dict[str, Any]):
     params = Params(replace_none(param_dict))
     prepare_environment(params)
 
-    log_dir = params.get("serialization_prefix", None)  # pylint: disable=no-member
+    trainer_params = params.pop("trainer")
+    log_dir = trainer_params.get("serialization_prefix")
```

Comment:
I mean, as long as `serialization_prefix` is allowed to be `None`, then it's not really an error when it ends up being `None`. Are we sure we want to allow that? What's the point of training without a serialization directory?